src/platform_impl/linux/wayland/window/mod.rs

```diff
@@ -417,115 +419,124 @@ impl Window {
 
     #[inline]
     pub fn set_theme<T: Theme>(&self, theme: T) {
-        let button_element_color =
-            |theme: &dyn Theme,
-             button_element_color: fn(&dyn Theme, ButtonState) -> [u8; 4]|
-             -> ButtonColorSpec {
-                let idle_color = button_element_color(theme, ButtonState::Idle);
-                let hovered_color = button_element_color(theme, ButtonState::Hovered);
-                let disabled_color = button_element_color(theme, ButtonState::Disabled);
-
-                let idle = ARGBColor {
-                    a: idle_color[0],
-                    r: idle_color[1],
-                    g: idle_color[2],
-                    b: idle_color[3],
-                };
-                let idle = ColorSpec {
-                    active: idle,
-                    inactive: idle,
-                };
-
-                let hovered = ARGBColor {
-                    a: hovered_color[0],
-                    r: hovered_color[1],
-                    g: hovered_color[2],
-                    b: hovered_color[3],
-                };
-                let hovered = ColorSpec {
-                    active: hovered,
-                    inactive: hovered,
-                };
-
-                let disabled = ARGBColor {
-                    a: disabled_color[0],
-                    r: disabled_color[1],
-                    g: disabled_color[2],
-                    b: disabled_color[3],
-                };
-                let disabled = ColorSpec {
-                    active: disabled,
-                    inactive: disabled,
-                };
-
-                ButtonColorSpec {
-                    idle,
-                    hovered,
-                    disabled,
-                }
-            };
-
-        let primary_element_color = |theme: &dyn Theme,
-                                     primary_element_color: fn(&dyn Theme, bool) -> [u8; 4]|
-         -> ColorSpec {
-            let active = primary_element_color(theme, true);
-            let inactive = primary_element_color(theme, false);
-
-            let active = ARGBColor {
-                a: active[0],
-                r: active[1],
-                g: active[2],
-                b: active[3],
-            };
-
-            let inactive = ARGBColor {
-                a: inactive[0],
-                r: inactive[1],
-                g: inactive[2],
-                b: inactive[3],
-            };
+        // First buttons is minimize, then maximize, and then close.
+        let buttons: Vec<(ButtonColorSpec, ButtonColorSpec)> =
+            [Button::Minimize, Button::Maximize, Button::Close]
+                .iter()
+                .map(|button| {
+                    let button = *button;
+                    let idle_active_bg = theme
+                        .button_color(button, false, true, ButtonState::Idle)
+                        .into_sctk();
```

Comment:
Couldn't this just be an `.into()` conversion? Or do you have it as explicitly typed `into_sctk` so you don't have to specify the target type since it can't be inferred?